### ISSUE TITLE
Add SHAP support

### ISSUE THREAD

**Issue by asperaa** [TeamHG-Memex/eli5#299]
@kmike @lopuhin Hi eli5 community,

I'm interested in working on the eli5 project. Specifically on the task of adding the SHAP support.
How should I start?

**Comment by lopuhin** [comment]
Hi @asperaa adding SHAP support is one of proposed Google Summer of Code projects, would you like to do it as a GSoC student, or independently?
In any case, I think a good way to start would be checking some existing small issues https://github.com/TeamHG-Memex/eli5/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%22 to get to know eli5 better.
Also there is already an issue about adding SHAP support here https://github.com/TeamHG-Memex/eli5/issues/240 - I think it's better to move any SHAP-specific discussion there

**Comment by asperaa** [comment]
I would like to do it as a GSoC student.

**Comment by asperaa** [comment]
Also, I will move the SHAP support discussions to #240.Much excited.

**Comment by lopuhin** [comment]
@asperaa great, thank you!

In order to get accepted, one needs

1) a good proposal, and
2) contribution to a project. 

(1) is needed to make sure you understand the project well, and have a concrete plan on how to implement it.
We ask for (2) to figure out how does it feel to work with you, and to ensure you're comfortable with the development process.

To write a good proposal I think you need to understand

1) what is eli5 library about, how it is used, how the code is organized, etc.;
2) SHAP algorithm (https://arxiv.org/abs/1802.03888) - what are its applications, how it works, what are its limitations;

So we suggest to try using eli5 for some problems, read its source code, check what's implemented, to understand where SHAP fits in - and also read SHAP paper, as well as try using https://github.com/slundberg/shap package.

After that it makes sense to check https://github.com/TeamHG-Memex/eli5 issue tracker (especially 
https://github.com/TeamHG-Memex/eli5/issues?q=is%3Aissue+is%3Aopen+label%3A%22help+wanted%2 but feel free to tackle any other issues you find), and try to fix some of the problems there, to get a contribution. You may also fix something which is not currently on the issue tracker, if you've found something when trying eli5.

**Comment by Sanyam07** [comment]
Hi , I also want to join . Can more than person join. I am already using eli5  and shap for the production purpose. But would like to combine them in one.

**Comment by Gallaecio** [comment]
@Sanyam07 I’m sorry, the student application deadline has passed already. See https://developers.google.com/open-source/gsoc/timeline

Maybe you can apply next year?
